src/AT_NumericalRoutines.c

```diff
@@ -718,3 +718,110 @@ double AT_get_interpolated_y_from_interval(const double left_x, const double lef
 		return  left_y + (right_y - left_y)*((intermediate_x - left_x) / (right_x - left_x));
 	}
 }
+
+
+void AT_get_interpolated_cubic_spline_y_tab_from_input_2d_table(const double input_data_xy[][2], const long lenght_of_input_data, const double intermediate_x[], const long lenght_of_intermediate_x_data,   double intermediate_y[]){
+  //auxiliary variables
+  double p, qn, sig, un;
+  //first derivative tab
+  double u[lenght_of_input_data];
+  //first derivative, points 0, lenght_of_input_data-1 boundary conditions
+  double  yp1, ypn;
+  //second derivative tab
+  double y2[lenght_of_input_data];
+
+  //set boundary variables
+
+  yp1 = (input_data_xy[1][1]-input_data_xy[0][1])/(input_data_xy[1][0]-input_data_xy[0][0]);
+  ypn = (input_data_xy[lenght_of_input_data-1][1]-input_data_xy[lenght_of_input_data-2][1])/(input_data_xy[lenght_of_input_data-1][0]-input_data_xy[lenght_of_input_data-2][0]);
```

Comment:
What will happen if input data will be a table with single element, i.e. `lenght_of_input_data` = 1 ? Will we get negative indices here ?

Reply:
See #100